src/components/OfflineSearch.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, Props, createElement } from "react";
+import { findDOMNode } from "react-dom";
+
+import * as dijitRegistry from "dijit/registry";
+
+import { Alert } from "./Alert";
+import "../ui/OfflineSearch.css";
+
+type searchMethodOptions = "equals" | "lessThan" | "lessThanOrEquals" | "greaterThan" | "greaterThanOrEquals" | "contains";
+
+interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: string | null;
+        _setsize: number;
+        _setSize: number;
+        atEnd: () => boolean;
+        _pageSize: number;
+    };
+    _loadMore: () => void;
+    _onLoad: () => void;
+    _renderData: () => void;
+    update: () => void;
+}
+
+interface OfflineSearchProps extends Props<OfflineSearch> {
+    defaultQuery: string;
+    searchEntity: string;
+    searchAttribute: string;
+    showSearchBar: string;
+    targetGridName: string;
+    searchMethod: searchMethodOptions;
+}
+
+interface OfflineSearchState {
+    alertMessage?: string;
+    buttonVisibility?: string;
+}
+
+class OfflineSearch extends Component<OfflineSearchProps, OfflineSearchState> {
+    // internal variables
+    private searchButton: HTMLButtonElement;
+    private searchInput: HTMLInputElement;
+    private targetWidget: ListView;
+
+    constructor(props: OfflineSearchProps) {
+        super(props);
+
+        this.state = {
+            alertMessage: "",
+            buttonVisibility: "hidden"
+        };
+        this.onClear = this.onClear.bind(this);
+        this.onSearchKeyDown = this.onSearchKeyDown.bind(this);
+    }
+
+    render() {
+        return createElement("div", {
+            className: `widget-offline-search ${this.props.showSearchBar}`
+        },
+            createElement(Alert, { message: this.state.alertMessage }),
+            createElement("div", { className: "search-container" },
+                createElement("span", { className: "glyphicon glyphicon-search" }),
+                createElement("input", {
+                    className: "form-control", placeholder: "Search",
+                    ref: input => this.searchInput = input as HTMLInputElement
+                }),
+                createElement("button", {
+                    className: `btn-transparent ${this.state.buttonVisibility}`,
+                    ref: button => this.searchButton = button as HTMLButtonElement
+                },
+                    createElement("span", { className: "glyphicon glyphicon-remove" })
+                )
+            )
+        );
+    }
+
+    componentDidMount() {
+        this.findTargetNode();
+        this.isValidWidget();
+        this.setUpEvents();
+        this.setDefaultSearch();
+    }
+
+    componentWillUnmount() {
+        this.searchButton.removeEventListener("click", this.onClear);
+        this.searchInput.removeEventListener("keyup", this.onSearchKeyDown);
+    }
+
+    private findTargetNode() {
```

Comment:
I have changed the name but i will not make the function static.
I am not making it static because `findDOMNode` finds the react dom using the instance of the component